internal/services/containers/kubernetes_cluster_other_resource_test.go

```diff
@@ -499,6 +499,47 @@ func TestAccKubernetesCluster_upgradeChannel(t *testing.T) {
 	})
 }
 
+func TestAccKubernetesCluster_nodeOsUpgradeChannel(t *testing.T) {
```

Comment:
We still have a test failure, it seems the casing on these needs to be updated - I also think the last step will cause an error when we pass in `"null"` as a string value, for the final step we need to use a config that omits the property entirely to test whether the provider correctly sets this to `none` when it's absent.

```
------- Stdout: -------
=== RUN   TestAccKubernetesCluster_nodeOsUpgradeChannel
=== PAUSE TestAccKubernetesCluster_nodeOsUpgradeChannel
=== CONT  TestAccKubernetesCluster_nodeOsUpgradeChannel
testcase.go:110: Step 1/8 error: Error running pre-apply refresh: exit status 1
Error: expected node_os_channel_upgrade to be one of [NodeImage SecurityPatch Unmanaged], got unmanaged
with azurerm_kubernetes_cluster.test,
on terraform_plugin_test.tf line 27, in resource "azurerm_kubernetes_cluster" "test":
27:   node_os_channel_upgrade   = "unmanaged"
--- FAIL: TestAccKubernetesCluster_nodeOsUpgradeChannel (2.06s)
FAIL
```

Reply:
all tests are now passing. although im unsure if the approach is viable since we might be enforcing the NodeImage value. Can you take a look?